src/frontegg-auth/src/auth.rs

```diff
@@ -180,123 +137,82 @@ impl Authentication {
         Ok(Some(Self::new(config, client, registry)))
     }
 
-    /// Exchanges a password for an access token and a refresh token.
-    pub async fn exchange_password_for_token(
+    /// Establishes a new authentication session.
+    ///
+    /// If successful, returns a handle to the authentication session.
+    /// Otherwise, returns the authentication error.
+    pub async fn authenticate(
         &self,
+        expected_email: &str,
         password: &str,
-        expected_email: String,
-    ) -> Result<ExchangePasswordForTokenResponse, Error> {
+    ) -> Result<AuthSessionHandle, Error> {
         let password: AppPassword = password.parse()?;
-        let req = ApiTokenArgs {
-            client_id: password.client_id,
-            secret: password.secret_key,
-        };
+        match self.authenticate_inner(expected_email, password).await {
+            Ok(handle) => {
+                trace!("authentication successful");
```

Comment:
Use `#[instrument(.., ret, err)]` on `authenticate_inner` to record Ok and Err returns should allow you to remove this whole match block.

Reply:
Ooh, fancy!